mediaire_toolbox/transaction_db/transaction_db.py

```diff
@@ -0,0 +1,117 @@
+from sqlalchemy.orm import sessionmaker
+
+from mediaire_toolbox.transaction_db.model import Transaction, create_all
+from mediaire_toolbox.task_state import TaskState
+
+import datetime
+
+
+class TransactionDB:
+    """Connection to a DB of transactions where we can track status, failures, 
+    elapsed time, etc."""
+
+    def __init__(self, engine):
+        """
+        Parameters
+        ----------
+        engine: SQLAlchemy engine
+        """
+        DBSession = sessionmaker(bind=engine)
+        self.session = DBSession()
+        create_all(engine)
+
+    def create_transaction(self, t: Transaction) -> int:
+        """will set the provided transaction object as queued, 
+        add it to the DB and return the transaction id."""
+        try:
+            t.task_state = TaskState.queued
+            self.session.add(t)
+            self.session.commit()
+            return t.transaction_id
+        except:
```

Comment:
Doesn't matter because I raise the exception again. This is just so that the connections are always properly released. Otherwise things could happen like the next time the session is used it fails because the connection expired.